Index: homeworks/lesson_2/l2_task1.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- homeworks/lesson_2/l2_task1.py	(date 1599766988804)
+++ homeworks/lesson_2/l2_task1.py	(date 1599766988804)
@@ -0,0 +1,6 @@
+"""
+1. Создать список и заполнить его элементами различных типов данных.
+Реализовать скрипт проверки типа данных каждого элемента.
+Использовать функцию type() для проверки типа.
+Элементы списка можно не запрашивать у пользователя, а указать явно, в программе.
+"""
Index: homeworks/lesson_2/l2_task2.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- homeworks/lesson_2/l2_task2.py	(date 1599767028212)
+++ homeworks/lesson_2/l2_task2.py	(date 1599767028212)
@@ -0,0 +1,6 @@
+"""
+2. Для списка реализовать обмен значений соседних элементов,
+т.е. Значениями обмениваются элементы с индексами 0 и 1, 2 и 3 и т.д.
+При нечетном количестве элементов последний сохранить на своем месте.
+Для заполнения списка элементов необходимо использовать функцию input().
+"""
Index: homeworks/lesson_2/l2_task3.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- homeworks/lesson_2/l2_task3.py	(date 1599767043799)
+++ homeworks/lesson_2/l2_task3.py	(date 1599767043799)
@@ -0,0 +1,5 @@
+"""
+3. Пользователь вводит месяц в виде целого числа от 1 до 12.
+Сообщить к какому времени года относится месяц (зима, весна, лето, осень).
+Напишите решения через list и через dict.
+"""
Index: homeworks/lesson_2/l2_task4.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- homeworks/lesson_2/l2_task4.py	(date 1599767061059)
+++ homeworks/lesson_2/l2_task4.py	(date 1599767061059)
@@ -0,0 +1,6 @@
+"""
+4. Пользователь вводит строку из нескольких слов, разделённых пробелами.
+Вывести каждое слово с новой строки.
+Строки необходимо пронумеровать.
+Если в слово длинное, выводить только первые 10 букв в слове.
+"""
Index: homeworks/lesson_2/l2_task5.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- homeworks/lesson_2/l2_task5.py	(date 1599767083353)
+++ homeworks/lesson_2/l2_task5.py	(date 1599767083353)
@@ -0,0 +1,11 @@
+"""
+5. Реализовать структуру «Рейтинг», представляющую собой не возрастающий набор натуральных чисел.
+У пользователя необходимо запрашивать новый элемент рейтинга.
+Если в рейтинге существуют элементы с одинаковыми значениями,
+то новый элемент с тем же значением должен разместиться после них.
+Подсказка. Например, набор натуральных чисел: 7, 5, 3, 3, 2.
+Пользователь ввел число 3. Результат: 7, 5, 3, 3, 3, 2.
+Пользователь ввел число 8. Результат: 8, 7, 5, 3, 3, 2.
+Пользователь ввел число 1. Результат: 7, 5, 3, 3, 2, 1.
+Набор натуральных чисел можно задать непосредственно в коде, например, my_list = [7, 5, 3, 3, 2].
+"""
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<project version=\"4\">\n  <component name=\"ProjectRootManager\" version=\"2\" project-jdk-name=\"Python 3.8 (GU_python_basics_07.09.2020) (2)\" project-jdk-type=\"Python SDK\" />\n</project>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/misc.xml	(revision 51f11316d903b2f3936a8de8468dca83b4aaa7b1)
+++ .idea/misc.xml	(date 1599766989131)
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.8 (GU_python_basics_07.09.2020) (2)" project-jdk-type="Python SDK" />
+  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.7 (venv)" project-jdk-type="Python SDK" />
 </project>
\ No newline at end of file
Index: homeworks/lesson_2/l2_task6.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- homeworks/lesson_2/l2_task6.py	(date 1599767231839)
+++ homeworks/lesson_2/l2_task6.py	(date 1599767231839)
@@ -0,0 +1,23 @@
+"""
+6. * Реализовать структуру данных «Товары». Она должна представлять собой список кортежей.
+Каждый кортеж хранит информацию об отдельном товаре.
+В кортеже должно быть два элемента — номер товара и словарь с параметрами
+(характеристиками товара: название, цена, количество, единица измерения).
+Структуру нужно сформировать программно, т.е. запрашивать все данные у пользователя.
+Пример готовой структуры:
+[
+(1, {“название”: “компьютер”, “цена”: 20000, “количество”: 5, “eд”: “шт.”}),
+(2, {“название”: “принтер”, “цена”: 6000, “количество”: 2, “eд”: “шт.”}),
+(3, {“название”: “сканер”, “цена”: 2000, “количество”: 7, “eд”: “шт.”})
+]
+Необходимо собрать аналитику о товарах. Реализовать словарь,
+в котором каждый ключ — характеристика товара, например название,
+а значение — список значений-характеристик, например список названий товаров.
+Пример:
+{
+“название”: [“компьютер”, “принтер”, “сканер”],
+“цена”: [20000, 6000, 2000],
+“количество”: [5, 2, 7],
+“ед”: [“шт.”]
+}
+"""
Index: .idea/GU_python_basics_07.09.2020.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<module type=\"PYTHON_MODULE\" version=\"4\">\n  <component name=\"NewModuleRootManager\">\n    <content url=\"file://$MODULE_DIR$\">\n      <excludeFolder url=\"file://$MODULE_DIR$/venv\" />\n    </content>\n    <orderEntry type=\"jdk\" jdkName=\"Python 3.8 (GU_python_basics_07.09.2020) (2)\" jdkType=\"Python SDK\" />\n    <orderEntry type=\"sourceFolder\" forTests=\"false\" />\n  </component>\n</module>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/GU_python_basics_07.09.2020.iml	(revision 51f11316d903b2f3936a8de8468dca83b4aaa7b1)
+++ .idea/GU_python_basics_07.09.2020.iml	(date 1599766989128)
@@ -4,7 +4,7 @@
     <content url="file://$MODULE_DIR$">
       <excludeFolder url="file://$MODULE_DIR$/venv" />
     </content>
-    <orderEntry type="jdk" jdkName="Python 3.8 (GU_python_basics_07.09.2020) (2)" jdkType="Python SDK" />
+    <orderEntry type="inheritedJdk" />
     <orderEntry type="sourceFolder" forTests="false" />
   </component>
 </module>
\ No newline at end of file
